packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -1,26 +1,58 @@
-import { Transformer, TransformerContext, InvalidDirectiveError, gql, getDirectiveArguments } from 'graphql-transformer-core'
+import { Transformer, TransformerContext, InvalidDirectiveError, gql, getDirectiveArguments, getFieldArguments } from 'graphql-transformer-core'
 import GraphQLAPI from 'cloudform-types/types/appSync/graphQlApi'
+import Resolver from 'cloudform-types/types/appSync/resolver';
 import { ResourceFactory } from './resources'
 import { AuthRule, ModelQuery, ModelMutation, ModelOperation } from './AuthRule'
 import {
     ObjectTypeDefinitionNode, DirectiveNode, ArgumentNode, TypeDefinitionNode, Kind,
-    FieldDefinitionNode, InterfaceTypeDefinitionNode
+    FieldDefinitionNode, InterfaceTypeDefinitionNode, valueFromASTUntyped,
 } from 'graphql'
-import { ResourceConstants, ResolverResourceIDs, isListType, getBaseType } from 'graphql-transformer-common'
+import { ResourceConstants, ResolverResourceIDs, isListType,
+    getBaseType, makeNamedType, makeInputValueDefinition,
+    makeNonNullType, graphqlName, toUpper, makeField } from 'graphql-transformer-common'
 import {
-    Expression, print, raw, iff, equals, forEach, set, ref, list, compoundExpression, or, newline,
+    Expression, print, raw, iff, forEach, set, ref, list, compoundExpression, or, newline,
     comment
 } from 'graphql-mapping-template';
-import { valueFromASTUntyped } from 'graphql'
 
 import {
-    OWNER_AUTH_STRATEGY,
     DEFAULT_OWNER_FIELD,
-    DEFAULT_IDENTITY_FIELD,
-    GROUPS_AUTH_STRATEGY,
-    DEFAULT_GROUPS_FIELD
+    ON_CREATE_FIELD,
+    ON_UPDATE_FIELD,
+    ON_DELETE_FIELD,
 } from './constants'
 
+interface QueryNameMap {
+    get?: string;
+    list?: string;
+    query?: string;
+}
+
+interface MutationNameMap {
+    create?: string;
+    update?: string;
+    delete?: string;
+}
+
+enum ModelSubscriptionLevel {
+    OFF = "OFF",
+    PUBLIC = "PUBLIC",
+}
+
+interface SubscriptionNameMap {
+    onCreate?: string[];
```

Comment:
With this structure you cannot configure level per operation. You can't make ```onCreate``` ```PUBLIC``` and protect only ```onUpdate``` and ```onDelete```. Are we ok with it?